doc/specs/stdlib_hash_maps.md

```diff
@@ -0,0 +1,3088 @@
+
+* `set( other, value )` - sets to content of other to value.
+
+* `free_other( other )` - frees the memory in other.
+
+Procedures to hash keys to 32 bit integers:
+
+* `fnv_1_hasher( key )` - hashes a key using the FNV-1 algorithm.
+
+* `fnv_1a_hasher( key )` - hashes a key using the FNV-1a algorithm.
+
+* `seeded_nmhash32_hasher( key )` - hashes a key using the nmhash32
+  algorithm.
+
+* `seeded_nmhash32x_hasher( key )` - hashes a key using the nmhash32x
+  algorithm.
+
+* `seeded_water_hasher( key )` - hashes a key using the waterhash
+  algorithm.
+
+### Specifications of the `stdlib_32_bit_key_data_wrapper` procedures
+
+#### `copy_key` - Returns a copy of the key
+
+##### Status
+
+Experimental
+
+##### Description
+
+Returns a copy of an input of type `key_type`
+
+##### Syntax
+
+`call [[stdlib_32_bit_key_data_wrapper:copy_key]]( key_in, key_out )`
+
+##### Class
+
+Subroutine.
+
+##### Arguments
+
+`key_in`: shall be a scalar expression of type `key_type`. It
+is an `intent(in)` argument.
+
+`key_out`: shall be a scalar variable of type `key_type`. It
+is an `intent(out)` argument.
+
+##### Example
+
+```fortran
+    program demo_copy_key
+      use stdlib_32_bit_key_data_wrapper, only: &
+          copy_key, key_test, key_type
+      use iso_fortran_env, only: int8
+      implicit none
+      integer(int8), allocatable :: value(:)
+      type(key_type) :: key_in, key_out
+      integer(int_8) :: i
+      allocate( value(1:15) )
+      do i=1, 15
+          value(i) = i
+      end do
+      call set( key, value )
+      call copy_key( key_in, key_out )
+      print *, "key_in == key_out = ", key_test( key_in, key_out )
+    end program demo_copy_key
+```
+
+#### `copy_other` - Returns a copy of the other data
+
+##### Status
+
+Experimental
+
+##### Description
+
+Returns a copy of an input of type `other_type`
+
+##### Syntax
+
+`call [[stdlib_32_bit_key_data_wrapper:copy_other]]( other_in, other_out )`
+
+##### Class
+
+Subroutine.
+
+##### Arguments
+
+`other_in`: shall be a scalar expression of type `other_type`. It
+is an `intent(in)` argument.
+
+`other_out`: shall be a scalar variable of type `other_type`. It
+is an `intent(out)` argument.
+
+##### Example
+
+```fortran
+    program demo_copy_other
+      use stdlib_32_bit_key_data_wrapper, only: &
+          copy_other, get, other_type, set
+      use iso_fortran_env, only: int8
+      implicit none
+      integer(int8), allocatable :: value1(:), value2(:)
+      type(other_type) :: other_in, other_out
+      integer(int_8) :: i
+      allocate( value1(1:15) )
+      do i=1, 15
+          value1(i) = i
+      end do
+      call set( other_in, value1 )
+      call copy_other( other_in, other_out )
+      call get( other_out, value2 )
+      print *, "other_in == other_out = ", &
+        all( value1 == value2 )
+    end program demo_copy_other
+```
+
+
+#### `FIBONACCI_HASH` - maps an integer to a smaller number of bits
```

Comment:
Is this different to the version in `stdlib_hash_32bit`?

Reply:
It is  just giving access to the version in `stdlib_hash_32bi`.

Reply:
If it is the case, then i suggest to to repeat the specs here. A link to the page of `Fibonacci_hash` should be enough,

Reply:
Do you mean "then i suggest not to repeat the specs here."?